src/storage.js

```diff
@@ -0,0 +1,430 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Privacy Badger.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+require.scopes.storage = (function() {
+
+
+/**
+ * # Storage Objects
+ *
+ * snitch_map is our collection of potential tracking base_domains.
+ * The key is a base domain (ETLD+1) and the value is an array of first
+ * party domains on which this tracker has been seen.
+ * it looks like this:
+ * { 
+ *   "third-party.com": ["a.com", "b.com", "c.com"], 
+ *   "eviltracker.net": ["eff.org", "a.com"]
+ * }
+ *
+ * action_map is where we store the action for each domain that we have 
+ * decided on an action for. Each subdomain gets its own entry. For example:
+ * {
+ *   "google.com": { heuristicAction: "block", dnt: false, userAction: ""}
+ *   "fonts.google.com": { heuristicAction: "cookieblock", dnt: false, userAction: ""}
+ *   "apis.google.com": { heuristicAction: "cookieblock", dnt: false, userAction: "user_block"}
+ *   "widget.eff.org": { heuristicAction: "block", dnt: true, userAction: ""}
+ * }
+ *
+ * cookieblock_list is where we store the current cookie block list as 
+ * downloaded from eff.org. The keys are the domains which should be blocked.
+ * The values are simply 'true'
+ *
+ * { 
+ *   "maps.google.com": true,
+ *   "creativecommons.org": true,
+ * }
+ **/
+
+// Cache of BadgerStorage objects
+var badgerPen = {};
+
+
+var initialize = function(callback){
+  console.log('loading badgers into the pen');
+  var storage_objects = [
+    "snitch_map",
+    "action_map",
+    "cookieblock_list",
+    "supercookie_domains",
+    "dnt_hashes",
+    "settings_map"
+  ];
+
+  _initializeCache(storage_objects, callback);
+};
+
+var getScore = function(action){
+  switch(action){
+    case pb.NO_TRACKING: return 0;
+    case pb.ALLOW: return 1;
+    case pb.BLOCK: return 2;
+    case pb.COOKIEBLOCK: return 3;
+    case pb.DNT: return 4;
+    default: return 5; 
+  }
+};
+
+/**
+ * get the current presumed action for a specific fqdn, ignoring any rules for subdomains
+ * below or above it
+ * @param {Object} domain domain object from action_map
+ * @returns {String} the presumed action for this FQDN
+ **/
+var getActionForFqdn = function(domain){
+  if(domain.userAction){ return domain.userAction; }
+  if(domain.dnt){ return pb.DNT; } 
+  if(domain.heuristicAction){ return domain.heuristicAction; } 
+  return pb.NO_TRACKING;
+};
+
+/**
+ * adds a heuristic action for a domain
+ * @param {String} domain Domain to add
+ * @param {String} action The heuristic action to take
+ */
+var setupHeuristicAction = function(domain, action){
+  _setupDomainAction(domain, action, "heuristicAction");
+};
+
+/**
+ * Sets up a domain for DNT
+ * @param {String} domain Domain to add
+ */
+var setupDNT = function(domain){
+  _setupDomainAction(domain, "true", "dnt"); 
```

Comment:
Should this set a boolean rather than a string? `_setupDomainAction(domain, true, "dnt");`